tests/integration/test_schema_coordinator.py

```diff
@@ -155,14 +170,41 @@ async def test_coordinator_workflow(
     secondary = coordinator if primary_selection_strategy == "highest" else coordinator2
     secondary_client = client if primary_selection_strategy == "highest" else client2
 
-    await wait_for_primary_state(primary)
-    assert not secondary.are_we_master
+    if primary == coordinator2:
+        # we need to disable the master for `waiting_time_before_acting_as_master_sec` seconds each time,
+        # a new node its elected as master.
+        # if the coordinator its `coordinator1` since isn't changed we don't have to wait
+        # for the `waiting_time_before_acting_as_master_sec` seconds.
+
+        # give time to the election to be forwarded to all the coordinators.
+        await asyncio.sleep(3)
+
+        assert (
+            not primary.are_we_master()
+        ), "after a change in the coordinator we can act as a master until we wait for the required time"
+        assert not secondary.are_we_master(), "also the second cannot be immediately a master"
+        # after that time the primary can act as a master
+        await asyncio.sleep(waiting_time_before_acting_as_master_sec)
+        assert not primary.are_we_master(), "Last fetch before being available as master"
+        assert not secondary.are_we_master(), "secondary cannot be master"
+
+    assert primary.are_we_master()
+    assert not secondary.are_we_master()
 
     # Check is closing the primary coordinator will rebalance the secondary to change to primary
     await primary.close()
     await primary_client.close()
 
-    await wait_for_primary_state(secondary)
+    now = time.time()
+    while time.time() - now < waiting_time_before_acting_as_master_sec:
+        assert not secondary.are_we_master(), (
+            f"Cannot become master before {waiting_time_before_acting_as_master_sec} seconds "
+            f"for the late records that can arrive from the previous master"
+        )
+        await asyncio.sleep(0.5)
+
+    while not secondary.are_we_master():
```

Comment:
This will loop until test timeout. Maybe this could abort earlier if secondary is not set in reasonable time?